doc/source/dev/contributor/rendering_documentation.rst

```diff
@@ -287,6 +287,43 @@ reproduce the results of the example exactly, so examples using random data
 should not refer to precise numerical values based on random data or rely on
 them to make their point.
 
+Legacy directive
+~~~~~~~~~~~~~~~~
+
+If a function, module or API is in *legacy* mode, meaning that it is kept around
+for backwards compatibility reasons, but is not recommended to use in new code,
+you can use the ``.. legacy::`` directive.
+
+By default, if used with no arguments, the legacy directive will generate the
+following output:
+
+.. legacy::
+
+
+If you need to include a custom message, such as a new API to replace the old
+one, this message will be appended to the default message::
```

Comment:
I would actually consider to force us to write something. That would help users have clear update paths.

If we don't want to force that, we could at least mention that it's recommended to do so.

Reply:
Let me know if this wording is ok.